build/esbuild-build.ts

```diff
@@ -43,7 +43,7 @@ esbuild
     ...esBuildContext,
     tsconfig: "tsconfig.web.json",
     platform: "browser",
-    outdir: "dist/esm",
+    outdir: "dist/esm/src",
```

Comment:
Why don't you output both in the same format? I guess cjs is the most common?

Reply:
> if I were to exclude the browser testing

By definition this code isn't polymorphic if you have to write separate tests for node and browser. You should make the code simpler to maintain by only writing one version that works in both environments.

Reply:
The difficulty I was having which forced my hand was dealing with the extraRpcs declaration not being included in the build with its data tho this amongst over things fixed that

Reply:
> Having distinct build outputs for ESM and CJS in a package is a common and recommended practice for maximizing compatibility and performance across different JavaScript environments. It requires careful configuration but ultimately makes your package more versatile and future-proof.

https://chat.openai.com/c/2180c787-0729-44ba-a247-332bc245c385